pkg/controller/logstash/reconcile.go

```diff
@@ -49,26 +51,34 @@ func reconcileStatefulSet(params Params, podTemplate corev1.PodTemplateSpec) (*r
 
 	var status logstashv1alpha1.LogstashStatus
 
-	if status, err = calculateStatus(&params, reconciled.Status.ReadyReplicas, reconciled.Status.Replicas); err != nil {
+	if status, err = calculateStatus(&params, reconciled); err != nil {
 		results.WithError(errors.Wrap(err, "while calculating status"))
 	}
 	return results, status
 }
 
 // calculateStatus will calculate a new status from the state of the pods within the k8s cluster
 // and will return any error encountered.
-func calculateStatus(params *Params, ready, desired int32) (logstashv1alpha1.LogstashStatus, error) {
+func calculateStatus(params *Params, sset appsv1.StatefulSet) (logstashv1alpha1.LogstashStatus, error) {
 	logstash := params.Logstash
 	status := params.Status
+	labelSelector := sset.Spec.Selector
 
 	pods, err := k8s.PodsMatchingLabels(params.Client, logstash.Namespace, map[string]string{NameLabelName: logstash.Name})
 	if err != nil {
 		return status, err
 	}
 
+	if labelSelector != nil {
+		selector, err := metav1.LabelSelectorAsSelector(labelSelector)
+		if err != nil {
+			return logstashv1alpha1.LogstashStatus{}, err
+		}
+		status.Selector = selector.String()
+	}
```

Comment:
nit: move the `labelSelector` variable closer to where it is used, I think we can omit it as well:

```suggestion
	pods, err := k8s.PodsMatchingLabels(params.Client, logstash.Namespace, map[string]string{NameLabelName: logstash.Name})
	if err != nil {
		return status, err
	}

	if sset.Spec.Selector != nil {
		selector, err := metav1.LabelSelectorAsSelector(sset.Spec.Selector)
		if err != nil {
			return logstashv1alpha1.LogstashStatus{}, err
		}
		status.Selector = selector.String()
	}
```